### PR TITLE
fix(Dropdown|Select|Form): fixes in typings

### LINKED ISSUE #1400: Modal, how to get access to onBlur method?
I'm using the modal in a slightly different way, controlling it's opening/closing using an event dispatcher. I'd like to use the native React event handler onBlur so that I can do so:
<Modal open={true} dimmer={false} onBlur={this.handleClose} ... >

private handleClose =  () => ModalWindowService.closeModal();

Of course, since I'm using TypeScript and I'm a n00b at it, I get this error:

ERROR in ./src/components/Modal/AssignBasketModal/AssignBasketModal.tsx
(34,55): error TS2339: Property 'onBlur' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<ModalProps, ComponentState>> & { childre...'.

I've realized that setting closeOnDocumentClick to true would do the same thing, however, I noticed this functionality doesn't work if the Modal open={true}

### PR DESCRIPTION
**Pull request by layershifter**
Fixes #1400.

### CODE REVIEW

File: src/addons/Select/index.d.ts [hunk 1]
```diff
@@ -8,7 +8,6 @@ import {
 } from '../../modules/Dropdown';
 
 export interface SelectProps extends DropdownProps {
-  selection: true;
```
Comment (layershifter) [inline src/addons/Select/index.d.ts:11]: `true` is not valid type, so it's cruft.

File: src/collections/Form/index.d.ts [hunk 2]
```diff
@@ -46,10 +46,6 @@ interface FormProps {
   widths?: 'equal';
 }
 
-interface FormOnSubmitData extends FormProps {
```
Comment (layershifter) [inline src/collections/Form/index.d.ts:49]: Cruft.

File: src/modules/Dropdown/index.d.ts [hunk 3]
```diff
@@ -106,71 +106,71 @@ export interface DropdownProps {
    * @param {SyntheticEvent} event - React's original SyntheticEvent.
    * @param {object} data - All props.
    */
-  onBlur: (event: React.KeyboardEvent<HTMLElement>, data: DropdownProps) => void;
+  onBlur?: (event: React.KeyboardEvent<HTMLElement>, data: DropdownProps) => void;
```
Comment (layershifter) [inline src/modules/Dropdown/index.d.ts:109]: Prop is optional

**Comment by levithomason** [comment]
Whoops, I missed that one on review as well.  Thanks for the update.

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.67`.

**Pull request merged**